### PR TITLE
Feat/web3 api phase B

### PR DESCRIPTION
**Pull request by amiryadid-fb**
## Pull Request Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes (link to the issue here)

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

- [x] Locally tested against Fireblocks API

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have added corresponding labels to the PR

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -217,7 +217,7 @@ export interface TransactionArguments {
     feePayerInfo?: TransactionArgumentsFeePayerInfo;
 }
 
-export enum SignerConnectionFeeLevel {
+export enum Web3ConnectionFeeLevel {
     HIGH = "HIGH",
     MEDIUM = "MEDIUM",
 }
```
Comment (idanya) [inline src/types.ts:223]: No `LOW`?

File: src/fireblocks-sdk.ts [hunk 2]
```diff
@@ -1148,19 +1152,58 @@ export class FireblocksSDK {
...
-        return await this.apiClient.issueGetRequest(`/v1/connections`);
+    private getWeb3ConnectionPath(type: Web3ConnectionType) {
+        const basePath = `/v1/connections`;
+
+        switch (type) {
+            case(Web3ConnectionType.WALLET_CONNECT): {
+                return `${basePath}/wc`;
+            }
+            default: {
+                return basePath;
```
Comment (idanya) [inline src/fireblocks-sdk.ts:1163]: returning the default is valid from the server side?

**Review by idanya — CHANGES_REQUESTED** [review]
Why do you need the function overloadings if the ENUM has only 1 entry?

File: src/types.ts (hunk 1 above)
Comment (amiryadid-fb) [reply src/types.ts:223]: Not on Wallet-Connect-Service

File: src/fireblocks-sdk.ts (hunk 2 above)
Comment (amiryadid-fb) [reply src/fireblocks-sdk.ts:1163]: Yeah it also routes to the service, will just be 404 for those methods

**Comment by amiryadid-fb** [comment]
> Why do you need the function overloadings if the ENUM has only 1 entry?

The method doesn't assume the type and I don't want it to be inferred that way for future editors of those methods, if someone else edits it in the future I want it to be clear this method is type agnostic, but I still want it to be friendly to users (so right now if someone uses it, he will see one call signature with the WALLET_CONNECT type and not the general ENUM)

File: src/fireblocks-sdk.ts (hunk 2 above)
Comment (amiryadid-fb) [reply src/fireblocks-sdk.ts]: Changed to throw error

**Review by idanya — APPROVED** [review]

**Review by yarinvak — APPROVED** [review]

**Pull request merged**